src/assets/Keycloak/oCIS-with-Keycloak.md

```diff
@@ -0,0 +1,194 @@
+      OCIS_URL: https://host.docker.internal:9200
+      OCIS_LOG_LEVEL: error
+      OCIS_INSECURE: true
+      OCIS_ADMIN_USER_ID: ""
+      OCIS_EXCLUDE_RUN_SERVICES: idp
+      GRAPH_ASSIGN_DEFAULT_USER_ROLE: false
+      GRAPH_USERNAME_MATCH: none
+      # PROXY
+      PROXY_TLS: false
+      PROXY_USER_OIDC_CLAIM: preferred_username
+      PROXY_USER_CS3_CLAIM: username
+    extra_hosts:
+      - host.docker.internal:host-gateway
+    labels:
+      traefik.enable: true
+      traefik.http.routers.ocis.tls: true
+      traefik.http.routers.ocis.rule: PathPrefix(`/`)
+      traefik.http.routers.ocis.entrypoints: ocis
+      traefik.http.services.ocis.loadbalancer.server.port: 9200
+
+  postgres:
+    image: postgres:alpine
+    volumes:
+      - postgres_data:/var/lib/postgresql/data
+    environment:
+      POSTGRES_DB: keycloak
+      POSTGRES_USER: keycloak
+      POSTGRES_PASSWORD: keycloak
+
+  keycloak:
+    image: quay.io/keycloak/keycloak:24.0.1
+    command: ["start-dev", "--proxy=edge"]
+    environment:
+      KC_DB: postgres
+      KC_DB_URL: "jdbc:postgresql://postgres:5432/keycloak"
+      KC_DB_USERNAME: keycloak
+      KC_DB_PASSWORD: keycloak
+      KC_FEATURES: impersonation
+      KEYCLOAK_ADMIN: admin
+      KEYCLOAK_ADMIN_PASSWORD: admin
+    extra_hosts:
+      - host.docker.internal:host-gateway
+    labels:
+      traefik.enable: true
+      traefik.http.routers.keycloak.tls: true
+      traefik.http.routers.keycloak.rule: PathPrefix(`/`)
+      traefik.http.routers.keycloak.entrypoints: keycloak
+      traefik.http.services.keycloak.loadbalancer.server.port: 8080
+    depends_on:
+      - postgres
+
+volumes:
+  postgres_data:
+```
+
+Save the above code in `compose.yaml` file locally and run docker command `docker compose up` in a terminal.
+
+Docker compose serves three services:
+- oCIS on `https://host.docker.internal:9200`
+- Keycloak on `https://host.docker.internal:8443`
+- Traefik dashboard on `http://host.docker.internal:8080/`
+
+### Manual Steps For Setting Up oCIS With Keycloak
+1. Go to oCIS url: `https://host.docker.internal:9200/`
+
+![oCIS](/src/assets/Keycloak/images/oCIS-Keycloak/oCIS-Keycloak-UI-setup-1.png)
+
+2. Go to Keycloak url: `https://host.docker.internal:8443/`
+
+![keycloak](/src/assets/Keycloak/images/oCIS-Keycloak/oCIS-Keycloak-UI-setup-2.png)
+
+3. Create new realm: `oCIS`
+
+![create realm](/src/assets/Keycloak/images/oCIS-Keycloak/create-realm-1.png)
+![create realm with name `oCIS`](/src/assets/Keycloak/images/oCIS-Keycloak/create-realm-2.png)
+> Note: the realm name should be the name provided in the oCIS environment variable: `OCIS_OIDC_ISSUER: https://host.docker.internal:8443/realms/oCIS`
+
+4. Add a new client:
+
+![client registration](/src/assets/Keycloak/images/oCIS-Keycloak/client-registration-1.png)
+
+- Client ID: `web`
+
+![add client](/src/assets/Keycloak/images/oCIS-Keycloak/client-registration-2.png)
+
+- Root URL: `https://host.docker.internal:9200`
```

Comment:
as you can see, the image keycloak has the keyword `Root URL`. I want to give readers a sense of similarities. Nothing much

Reply:
What about using `URL` then throughout your blog? Or is the UI itself inconsistent having other instances where `url` is used in the UI?

Reply:
I want to keep a small url.

Reply:
What about opening an issue to point this out to them? It's no big thing but it would improve the User Experience. Maybe they gonna fix the Keycloak UI in the future :)

Reply:
in future when I will free. I will make it, right now my hands are full of tasks. Let's not block this blog for such thing which is not under my control

Reply:
only takes 2 minutes :)

Reply:
https://github.com/keycloak/keycloak/issues/new?assignees=&labels=kind%2Fenhancement%2Cstatus%2Ftriage&projects=&template=enhancement.yml

Reply:
as i repeat please lets not blog this PR for this topic

Reply:
It's a suggestion about living out our value of quality, and it's up to you if you want to help Keyclock to improve their user experience and overall quality. According to our company values, we are committed to constantly improve the quality in every aspect of our work.